benchmarking/frameworks/framework_base.py

```diff
@@ -103,6 +109,36 @@ def runBenchmark(self, info, benchmark, platform):
                 platform.delFilesFromPlatform(input_files)
         return output, output_files
 
+    @abc.abstractmethod
+    def composePreprocessCommand(self, preprocess_info):
+        files_db = {'preprocess': {'files': {}}}
+        for f_key in preprocess_info["files"]:
+            f_value = preprocess_info["files"][f_key]
+            if 'location' in f_value:
+                files_db['preprocess']['files'][f_key] = f_value['location']
+            if 'target' in f_value:
+                files_db['preprocess']['files'][f_key] = f_value['target']
+
+        assert 'program' in files_db['preprocess']['files'], "No program in preprocess"
+        program = files_db['preprocess']['files']['program']
+
+        # parse arguments
+        commands = []
+        arguments = preprocess_info["arguments"].split()
+        pattern = re.compile("\{([\w|\.]+)\}")
```

Comment:
can we consolidate the code to use the same pattern matching algorithm as the commands? Right now it duplicates code and may not be general enough. What do you think? 

Just extract line 148-168 to a function, and call that function here also.

Reply:
Planned to do. Haven't fully understood your code. :)